### PR TITLE
Delay endpoint setup for Spring

### LINKED ISSUE #14: HttpServlet does not have logStart for Spring
This would require a bit of research in how to make the servlet aware of HandlerInterceptors or use inheritance in the filters. Currently for /lbClassicStatus and for /health/outstanding there's no logStart.

### PR DESCRIPTION
**Pull request by maheshgaya**
- [x] I have followed the [Contribution Guidelines](../blob/master/.github/CONTTRIBUTING.md).
- [x] My code follows the code style of this project.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed. (Has passed ```mvn clean verify``` locally.)
- [x] pom.xml version follows the [Semantic Versioning](https://semver.org/) rules.
- [x] I have updated the documentation as necessary.
- [x] I have added myself to the [Contributors File](../blob/master/CONTRIBUTORS.md).
- [x] I know all contributors must sign the Contributor License Agreement.

Changes in this pr:
- Allow customization of WorkSummaryServlet. Allowing client apps to have endpoints instead of service for outstanding page. SpringWorkSummaryServlet provides an implementation for that.
- Delay endpoint setup for requests that go through Spring lifecycle
- Setup endpoint immediately for Servlets (this will be first step for fixing #14)
- Added logStart for Servlet fixes #14

### CODE REVIEW

File: work-tracker-examples/spring-example/src/main/webapp/WEB-INF/web.xml [hunk 1]
```diff
@@ -91,7 +91,7 @@ limitations under the License.
 
     <servlet>
         <servlet-name>workHttpServlet</servlet-name>
-        <servlet-class>com.deere.isg.worktracker.servlet.WorkHttpServlet</servlet-class>
+        <servlet-class>com.deere.isg.worktracker.spring.SpringWorkHttpServlet</servlet-class>
```
Comment (AveryRegier) [inline work-tracker-examples/spring-example/src/main/webapp/WEB-INF/web.xml]: We need to document the need for this change by applications in a release note.

File: work-tracker-spring/src/main/java/com/deere/isg/worktracker/spring/ServletEndpointRegistry.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+class ServletEndpointRegistry {
+    private static EndpointTrie trie = new EndpointTrie();
+
+    public static void populate(ServletContext context) {
+        if (context == null){
+            return;
+        }
+
+        context.getServletRegistrations()
+                .forEach((servletName, registration) -> addMappings(registration));
```
Comment (AveryRegier) [inline work-tracker-spring/src/main/java/com/deere/isg/worktracker/spring/ServletEndpointRegistry.java]:
```java
context.getServletRegistrations().valueSet().forEach((this::addMappings);
```

File: work-tracker-spring/src/main/java/com/deere/isg/worktracker/spring/ServletEndpointRegistry.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+                return;
+            }
+
+            EndpointTrie current = this;
+            for (char ch: value.toCharArray()){
+                if (!current.map.containsKey(ch)) {
+                    current.map.put(ch, new EndpointTrie());
+                }
+                current = current.map.get(ch);
+            }
```
Comment (AveryRegier) [inline work-tracker-spring/src/main/java/com/deere/isg/worktracker/spring/ServletEndpointRegistry.java]: These for loops over the characters are weird. What are they accomplishing?  Can it be extracted and named appropriately to help readability?

File: work-tracker-spring/src/main/java/com/deere/isg/worktracker/spring/ServletEndpointRegistry.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        if (context == null){
+            return;
+        }
+
+        context.getServletRegistrations()
+                .forEach((servletName, registration) -> addMappings(registration));
+    }
+
+    public static boolean contains(String value){
+        return trie.isEqual(value);
```
Comment (AveryRegier) [inline work-tracker-spring/src/main/java/com/deere/isg/worktracker/spring/ServletEndpointRegistry.java]: Is there a reason we don't just use a ConcurrentHashSet?  It is hard to find but it does exist:

```java
Set foo = ConcurrentHashMap.newKeySet();
```

File: work-tracker-spring/src/main/java/com/deere/isg/worktracker/spring/ServletEndpointRegistry.java [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+    static void clear(){
+        trie = new EndpointTrie();
+    }
+
+    private static void addMappings(ServletRegistration registration) {
+        registration.getMappings().forEach(ServletEndpointRegistry::populate);
+    }
+
+    private static class EndpointTrie{
```
Comment (AveryRegier) [inline work-tracker-spring/src/main/java/com/deere/isg/worktracker/spring/ServletEndpointRegistry.java]: Just extracting this class to a util package somewhere would at least make it _seem_ less complex.

File: work-tracker-spring/src/main/java/com/deere/isg/worktracker/spring/ServletEndpointRegistry.java (hunk 4 above)
Comment (maheshgaya) [reply work-tracker-spring/src/main/java/com/deere/isg/worktracker/spring/ServletEndpointRegistry.java]: I updated the endpointTrie implementation with set. Didn't know if you could do that from concurrentHashMap

File: work-tracker-spring/src/main/java/com/deere/isg/worktracker/spring/ServletEndpointRegistry.java (hunk 3 above)
Comment (maheshgaya) [reply work-tracker-spring/src/main/java/com/deere/isg/worktracker/spring/ServletEndpointRegistry.java]: Refactor this into Set

File: work-tracker-spring/src/main/java/com/deere/isg/worktracker/spring/ServletEndpointRegistry.java (hunk 5 above)
Comment (maheshgaya) [reply work-tracker-spring/src/main/java/com/deere/isg/worktracker/spring/ServletEndpointRegistry.java]: Refactor this into Set

**Pull request merged**